### PR TITLE
[enhancement](JNI) Provide default environment variables if it is unset

### PR DESCRIPTION
**Pull request by zy-kkk**
## Proposed changes

Issue Number: close #xxx

from #19041 
BE with java support needs some environment variables to start up and we have set them up in the start scripts (such as `start_be.sh`). However, if we debug BE directly, these environment variables are not set by default. For the convenience, we should provide default environment variables under this scenario.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zy-kkk** [comment]
run buildall

File: be/src/util/jni-util.cpp [hunk 1]
```diff
@@ -80,15 +112,21 @@ const std::string GetDorisJNIClasspathOption() {
...
+            "{}/conf:{}:{}", doris_home, GetDorisJNIDefaultClasspath(), original_classpath);
     setenv("CLASSPATH", classpath.c_str(), 0);
 
+    //LD_LIBRARY_PATH
+    std::string original_ld_library_path =
+            getenv("LD_LIBRARY_PATH") ? getenv("LD_LIBRARY_PATH") : "";
+    std::string new_ld_library_path =
+            fmt::format("{}/lib/hadoop_hdfs/native:{}", doris_home, original_ld_library_path);
+    setenv("LD_LIBRARY_PATH", new_ld_library_path.c_str(), 0);
+
```
Comment (adonis0147) [inline be/src/util/jni-util.cpp:126]: It is not suitable to use the environment `LD_LIBRARY_PATH`. This environment is usually used for test. You would better to use `-Djava.library.path` like the original snippet.

**Review by adonis0147 — CHANGES_REQUESTED** [review]

File: be/src/util/jni-util.cpp [hunk 2]
```diff
@@ -49,17 +49,31 @@ const std::string GetDorisJNIDefaultClasspath() {
...
+                if (!out.str().empty()) {
+                    out << ":";
+                }
+                out << entry.path().string();
+            }
         }
+    };
+
+    addJarsFromPath(std::string(doris_home) + "/lib");
+    addJarsFromPath(std::string(doris_home) + "/custom_lib");
```
Comment (adonis0147) [inline be/src/util/jni-util.cpp:68]:
```suggestion
    add_jars_from_path(std::string(doris_home) + "/lib");
    add_jars_from_path(std::string(doris_home) + "/custom_lib");
```

File: be/src/util/jni-util.cpp [hunk 3]
```diff
@@ -49,17 +49,31 @@ const std::string GetDorisJNIDefaultClasspath() {
     DCHECK(doris_home) << "Environment variable DORIS_HOME is not set.";
 
     std::ostringstream out;
-    std::string path(doris_home);
-    path += "/lib";
-    for (const auto& entry : std::filesystem::recursive_directory_iterator(path)) {
-        if (entry.path().extension() != ".jar") {
-            continue;
+
+    auto addJarsFromPath = [&](const std::string& base_path) {
```
Comment (adonis0147) [inline be/src/util/jni-util.cpp:53]:
```suggestion
    auto add_jars_from_path = [&](const std::string& base_path) {
```

File: be/src/util/jni-util.cpp [hunk 4]
```diff
@@ -49,17 +49,31 @@ const std::string GetDorisJNIDefaultClasspath() {
...
+            }
         }
+    };
+
+    addJarsFromPath(std::string(doris_home) + "/lib");
+    addJarsFromPath(std::string(doris_home) + "/custom_lib");
+
+    // Check and add HADOOP_CONF_DIR if it's set
+    const auto* hadoop_conf_dir = getenv("HADOOP_CONF_DIR");
+    if (hadoop_conf_dir && *hadoop_conf_dir) {
```
Comment (adonis0147) [inline be/src/util/jni-util.cpp:72]:
```suggestion
    if (hadoop_conf_dir != nullptr && strlen(hadoop_conf_dir) > 0) {
```

File: be/src/util/jni-util.cpp [hunk 5]
```diff
@@ -80,15 +94,17 @@ const std::string GetDorisJNIClasspathOption() {
...
+            "{}/conf:{}:{}", doris_home, GetDorisJNIDefaultClasspath(), original_classpath);
     setenv("CLASSPATH", classpath.c_str(), 0);
 
     // LIBHDFS_OPTS
-    setenv("LIBHDFS_OPTS",
-           fmt::format("-Djava.library.path={}/lib/hadoop_hdfs/native", getenv("DORIS_HOME"))
-                   .c_str(),
-           0);
+    const std::string java_opts = getenv("JAVA_OPTS") ? getenv("JAVA_OPTS") : "";
+    std::string libhdfs_opts = fmt::format("{} -Djava.library.path={}/lib/hadoop_hdfs/native",
```
Comment (adonis0147) [inline be/src/util/jni-util.cpp:104]: Should add the path `DORIS_HOME/lib` for `libz.so`.

**Review by adonis0147 — CHANGES_REQUESTED** [review]

**Review by adonis0147 — APPROVED** [review]

**Comment by zy-kkk** [comment]
run buildall

**Comment by zy-kkk** [comment]
run external

**Comment by zy-kkk** [comment]
run buildall

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**